presto-hive/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java

```diff
@@ -0,0 +1,354 @@
+        result.setDbName(partition.getDatabaseName());
+        result.setTableName(partition.getTableName());
+        result.setValues(partition.getValues());
+        result.setSd(makeStorageDescriptor(partition.getTableName(), partition.getColumns(), partition.getStorage()));
+        result.setParameters(partition.getParameters());
+        return result;
+    }
+
+    public static Table fromMetastoreApiTable(org.apache.hadoop.hive.metastore.api.Table table)
+    {
+        StorageDescriptor storageDescriptor = table.getSd();
+        if (storageDescriptor == null) {
+            throw new PrestoException(HIVE_INVALID_METADATA, "Table is missing storage descriptor");
+        }
+
+        Table.Builder tableBuilder = Table.builder()
+                .setDatabaseName(table.getDbName())
+                .setTableName(table.getTableName())
+                .setOwner(table.getOwner())
+                .setTableType(table.getTableType())
+                .setDataColumns(storageDescriptor.getCols().stream()
+                        .map(MetastoreUtil::fromMetastoreApiFieldSchema)
+                        .collect(Collectors.toList()))
+                .setPartitionColumns(table.getPartitionKeys().stream()
+                        .map(MetastoreUtil::fromMetastoreApiFieldSchema)
+                        .collect(Collectors.toList()))
+                .setParameters(table.getParameters() == null ? ImmutableMap.of() : table.getParameters())
+                .setViewOriginalText(makeOptional(table.getViewOriginalText()))
+                .setViewExpandedText(makeOptional(table.getViewExpandedText()));
+
+        fromMetastoreApiStorageDescriptor(storageDescriptor, tableBuilder.getStorageBuilder(), table.getTableName());
+
+        return tableBuilder.build();
+    }
+
+    public static Partition fromMetastoreApiPartition(org.apache.hadoop.hive.metastore.api.Partition partition)
+    {
+        StorageDescriptor storageDescriptor = partition.getSd();
+        if (storageDescriptor == null) {
+            throw new PrestoException(HIVE_INVALID_METADATA, "Partition does not contain a storage descriptor: " + partition);
+        }
+
+        Partition.Builder partitionBuilder = Partition.builder()
+                .setDatabaseName(partition.getDbName())
+                .setTableName(partition.getTableName())
+                .setValues(partition.getValues())
+                .setColumns(storageDescriptor.getCols().stream()
+                        .map(MetastoreUtil::fromMetastoreApiFieldSchema)
+                        .collect(Collectors.toList()))
+                .setParameters(partition.getParameters());
+
+        fromMetastoreApiStorageDescriptor(storageDescriptor, partitionBuilder.getStorageBuilder(), format("%s.%s", partition.getTableName(), partition.getValues()));
+
+        return partitionBuilder.build();
+    }
+
+    private static Properties getHiveSchema(
```

Comment:
It would be easier to read if this method were immediately below the two callers above `getHiveSchema`